libs/traits/src/connectors.rs

```diff
@@ -21,19 +21,31 @@ pub trait Codec: Sized {
 	fn deserialize<I: Input>(input: &mut I) -> Result<Self, codec::Error>;
 }
 
-/// The trait required for processing outgoing connectors messages.
+/// The trait required for sending outbound messages.
+pub trait Router {
+	/// The sender type of the outbound message.
+	type Sender;
+
+	/// The outbound message type.
+	type Message;
+
+	/// Send the message to the router's destination.
+	fn send(&self, sender: Self::Sender, message: Self::Message) -> DispatchResult;
+}
+
+/// The trait required for processing outbound connectors messages.
 pub trait OutboundQueue {
 	/// The sender type of the outgoing message.
 	type Sender;
 
-	/// The connector message enum.
+	/// The message type that is processed.
 	type Message;
 
 	/// The destination this message should go to.
 	type Destination;
 
 	/// Submit a message to the outbound queue.
-	fn submit(
+	fn handle(
```

Comment:
Sorry, to chip in here, but I don't like the naming of `handle`. I would assume that handle in most cases means a direct processing, where a queue in most cases firstly receives something and then acts when it has capacity for it. But no blocker.